executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -106,6 +86,10 @@ impl<'a, T: FieldElement> BlockMachineProcessor<'a, T> {
             }
         }
 
+        if witgen.has_unsolved_assignments() {
+            return Err("Unsolved assignments".to_string());
```

Comment:
So compared to before, we loose the information which assignments are not solvable, which I think could be quite useful for debugging! If you extract the `Assignment` struct, you could implement `Display` on it and add it to the error message.